server/pkg/webserver/middleware/auth.go

```diff
@@ -0,0 +1,59 @@
+package middleware
+
+import (
+	"github.com/iris-contrib/middleware/jwt"
+	"github.com/kataras/iris/v12"
+)
+
+var (
+	mySecret     = []byte("login secret")
+	WhiteListMap = map[string]struct{}{
+		"POST/api-nebula/db/connect": {},
+	}
+)
+
+func GetLoginTokenHandler(nebulaAddress string, username string) string {
+	token := jwt.NewTokenWithClaims(jwt.SigningMethodHS256, jwt.MapClaims{
+		"nebulaAddress": nebulaAddress,
+		"username":      username,
+	})
+	tokenString, _ := token.SignedString(mySecret)
+
+	return tokenString
+}
+
+func AuthenticatedLoginHandler(ctx iris.Context) error {
+	url := ctx.RouteName()
+	//HACK: Whitelisted urls do not require JWT authentication
+	if _, ok := WhiteListMap[url]; ok {
+		ctx.Next()
+		return nil
+	}
+
+	j := jwt.New(jwt.Config{
+		ValidationKeyGetter: func(token *jwt.Token) (interface{}, error) {
+			return mySecret, nil
+		},
+		Expiration:    false,
```

Comment:
Will it never expire? Is it possible to set an expiration time?

Reply:
I have set it.